### PR TITLE
Adds checking of reserved keywords against team names

### PR DESCRIPTION
**Pull request by ghost**
This fixes and solves gogit/gogs#3789

### CODE REVIEW

**Comment by tboerger** [comment]
LGTM but needs to be rebased

**Comment by tboerger** [comment]
Please reopen against master.

**Comment by strk** [comment]
Would it make sense for the empty name case to be handled by the IsUsableTeamName directly ?
Also, it looks like the IsUsableTeamName function is a perfect candidate for an automated test, how about adding an models/org_team_test.go  ?

**Comment by strk** [comment]
Full link to original issue: https://github.com/gogits/gogs/issues/3789

**Comment by thibaultmeyer** [comment]
I agree with @strk , if the team name is empty, don't enter on `IsUsableTeamName` function. If empty, you are sure at 100% it was an invalid team name, no need to check if is forbidden or not.

Maybe forbidden team names could be move to `app.ini` file (or a special file just for this purpose) or into database to allow users to edit list without recompile project.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by tboerger** [comment]
> Maybe forbidden team names could be move to app.ini file (or a special file just for this purpose) or into database to allow users to edit list without recompile project.

This list contains only names that are forbidden to prevent issues with the routes.

File: models/org_team.go
Comment (tboerger) [inline models/org_team.go]: Yes, the others are right, move that below the `if len(t.Name) == 0 {` check.

**Review by tboerger — CHANGES_REQUESTED** [review]

File: models/org_team.go
Comment (strk) [reply models/org_team.go]: My suggestion was actually to move the `len(t.Name) == 0` check inside `IsUsableTeamName` (as empty is clearly not a usable team name)

File: models/org_team.go
Comment (ghost) [reply models/org_team.go:201]: I did it in the same manner as `models/user.go#L504:L508`.

**Comment by ghost** [comment]
@tboerger Fixed.

File: models/org_team.go
Comment (bkcsoft) [reply models/org_team.go:201]: Macaron isn't (apparently) case insensitive ( test by going to `/org/CREATE` instead of `/org/create` )

**Review by tboerger — APPROVED** [review]

File: models/org_team.go
Comment (tboerger) [reply models/org_team.go:201]: @lunny is right, you should always downcase the name.

**Comment by strk** [comment]
LGTM

File: models/org_team.go
Comment (tboerger) [inline models/org_team.go:200]:
```
for _, reservedName := range reservedTeamNames {
  if reservedName == strings.ToLower(name) {

  }
}
```

**Review by tboerger — CHANGES_REQUESTED** [review]

**Comment by tboerger** [comment]
Sorry, but after seeing the comment of @lunny I changed my opinion :D

**Review by lunny — APPROVED** [review]

**Comment by strk** [comment]
@tboerger changed your mind so you want to `Dismiss review` ?
@LefsFlarey we still need tests, according to codecov (and it'd be worth a rebase)

**Comment by strk** [comment]
Assuming everyone is happy with this, merging (since github lets me)

**Comment by tboerger** [comment]
I didn't realized that before, but why have that been merged when there are still requests for changes left? I requested to do always a downcase check...

**Pull request merged**
